src/main/scala-2.12/org/ergoplatform/appkit/ergotool/AssetsAtomicExchangeMatchCmd.scala

```diff
@@ -76,22 +75,28 @@ case class AssetsAtomicExchangeMatchCmd(toolConf: ErgoToolConfig,
       if (!buyerHolderBox.getErgoTree.constants.contains(SigmaPropConstant(buyerAddress.getPublicKey))) {
         error(s"cannot find buyer's address $buyerAddress in buyer contract in box $buyerHolderBoxId")
       }
-      if (!sellerHolderBox.getErgoTree.constants.contains(LongConstant(buyerHolderBox.getValue))) {
-        error(s"cannot find token price ${buyerHolderBox.getValue}(from buyerHolderBox.value) in seller contract in box $sellerHolderBoxId")
-      }
+      // TODO buyerHolderBox.value may be greater or equal
+//      if (!sellerHolderBox.getErgoTree.constants.contains(LongConstant(buyerHolderBox.getValue))) {
+//        error(s"cannot find token price ${buyerHolderBox.getValue}(from buyerHolderBox.value) in seller contract in box $sellerHolderBoxId")
+//      }
       val token = sellerHolderBox.getTokens.get(0)
       if (!buyerHolderBox.getErgoTree.constants.contains(ByteArrayConstant(token.getId.getBytes))) {
         error(s"cannot find token id ${token.getId} in buyer contract in box $buyerHolderBoxId")
       }
      if (!buyerHolderBox.getErgoTree.constants.contains(LongConstant(token.getValue))) {
         error(s"cannot find token amount ${token.getValue} in buyer contract in box $buyerHolderBoxId")
       }
-      val boxesForTxFee = BoxOperations.selectTop(unspent, MinFee)
-      boxesForTxFee.addAll(util.Arrays.asList(buyerHolderBox, sellerHolderBox))
-      val inputBoxes = boxesForTxFee
+      val ergAmount = sellerHolderBox.getErgoTree.constants(6).asInstanceOf[Constant[SLong.type]].value
+
+      val claimableValue = buyerHolderBox.getValue - ergAmount + sellerHolderBox.getValue - 1 // 1 for buyerTokensOutBox.value
+      val dexFee = claimableValue - MinFee
+      if (dexFee <= minDexFee) {
+        error(s"found DEX fee = (claimable value - miner's fee) to be $dexFee, which is <= minDexFee ")
+      }
+      val inputBoxes = util.Arrays.asList(buyerHolderBox, sellerHolderBox)
       val txB = ctx.newTxBuilder
       val buyerTokensOutBox = txB.outBoxBuilder
-        .value(sellerHolderBox.getValue)
+        .value(1)
```

Comment:
MinFee is minimal value that can be put in the box. 1 NanoErg is too small.
See org.ergoplatform.appkit.Parameters

Reply:
Thanks. I missed that here. Fixed in https://github.com/aslesarenko/ergo-tool/pull/3/commits/424752c24d123e7cfe7d18124667a64f7d087d76